docs/design/baremetal/networking-infrastructure.md

```diff
@@ -70,7 +70,101 @@ DNS requirements internal to the cluster as possible, leaving only a
 small amount of public DNS configuration to be implemented by the user
 before starting the installation process.
 
-TODO - explain how this works ...
+In a `baremetal` environment, we do not know the IP addresses of all hosts in
+advance.  Those will come from an organization’s DHCP server.  Further, we can
+not rely on being able to program an organization’s DNS infrastructure in all
+cases.  We address these challenges in two ways:
 
-Explain that $cluster_name and $base_domain are supplied via
-install-config.yaml
+1. Self host some DNS infrastructure to provide DNS resolution for records only
+   needed internal to the cluster.
+2. Make use of mDNS (Multicast DNS) to dynamically discover the addresses of
+   hosts that we must resolve records for.
+
+### mdns-publisher
+
+https://github.com/openshift/mdns-publisher
+
+The `mdns-publisher` is the component that runs on each host to make itself
+discoverable by other hosts in the cluster.  It will be run by the
+`machine-config-operator`, once the following PR is complete:
```

Comment:
it is not run on the bootstrap node by the installer because ...

Reply:
Added:

```
`mdns-publisher` is not run on the bootstrap node, as there is no need for any
other host to discover the IP address that the bootstrap VM gets from DHCP.                                                                                                                                                                             
```